### DISCUSSION TITLE
Dynamic component using checkbox v-model changed after click outside

### DISCUSSION THREAD

**Discussion by tarikrital** [quasarframework/quasar#14432] [General - Components / Directives / etc] [answered]
Using a `q-checkbox` within an item of `q-uploader` list attachments get `v-model` updated after clicking outside the `q-uploader`, any info about the reason.

**Comment by MilosPaunovic** [comment]
Provide a CodePen reproduction first for this to be properly investigated.

**Reply by tarikrital** [reply]
In the codepen the model value doesn't change! 

https://codepen.io/wpplumber/pen/GRdmxdW

**Reply by MilosPaunovic** [reply]
Then I assume it has something to do with other bits of your code. Try to separate that logic and figure out is it really Quasar related.

**Reply by tarikrital** [reply]
I used an html input checkbox and it works see the codepen again which means related to Quasar and I also found a similsr issue for [vuetify](https://github.com/vuetifyjs/vuetify/issues/10849#issuecomment-1124517741).

**Reply by pdanpdan** [reply]
Can you please make a codepen that can reproduce the problem, explain what is the problem, and give step by step instructions on how to reproduce?

What would you expect this to do?

```html
<q-checkbox label="Expirable" v-model="file.expirable" :val="file.expirable"></q-checkbox>
```

**Reply by tarikrital** [reply]
I expect when clicked to change the state as with html checkbox!

**Comment by pdanpdan** [comment]
What state and with what value?

**Reply by tarikrital** [reply]
I made a fake Attachments uncomment the q-uploader or the q-checkbox to see how the checkbox behaves?

**Comment by pdanpdan** [comment]
This is your codepen just with file.expirable displayed
https://codepen.io/pdanpdan/pen/vYjZEad

**Reply by tarikrital** [reply]
The `q-checkbox` doesn't change the state after click but html checkbox do!

![image](https://user-images.githubusercontent.com/52907282/191128362-8350eb6e-ffac-4b65-9e66-7d163b9c5bbb.png)

**Comment by pdanpdan** [comment]
Yes, it is expected and normal as the value of the model does not change or it is not reactive

**Reply by tarikrital** [reply]
After **removing** the value the same result, I remind myself and you to keep answers on the same thread.

**Comment by pdanpdan** [comment]
You don't understand. Can you make the debug print of file.expirable change by clicking on the native checkbox?

**Comment by tarikrital** [comment, marked as answer]
I got it fixed by using  `Object.assign` on `@click` of `q-checkbox`.
